config/option.go

```diff
@@ -223,6 +223,10 @@ type Option struct {
 	// ValidationRegex is considered immutable after the option has
 	// been created.
 	ValidationRegex string
+	// ValidationFunc may contain a function to validate more complex values.
+	// The error is returned beyond the scope of this package and may be
+	// displayed to a user.
+	ValidationFunc func(value interface{}) error `json:"-"`
```

Comment:
I'd rather define a error type that should be returned here so we can be sure to have meaning full error messages for the user using a defined format.
Also, a note on the value type could be added because I guess the value is the one from `json.Unmarshal` so users of this function should known which types the can expect to receive (i.e. no `[]strng` but always `[]interface{}`)

Reply:
Maybe it would be even better to pass the raw `json.RawMessage` here?

Reply:
While I saw below how the values are decoded and passed I still feel like that should be part of the comment.